### ISSUE TITLE
Additional columns in Griddle 0.5.1 - data below not showing

### ISSUE THREAD

**Issue by jonifen** [GriddleGriddle/Griddle#391]
I don't know if this was originally an unintended piece of functionality, or if it's broken now, but adding extra columns to an instance of Griddle no longer works in the current new version (0.5.1).
I get a JSON response from an API which contains 5 fields, all of which are set to be displayed in the grid. However, I also need an extra column to be shown to include some other information, so I added another column to the columns array with a random name, and added the corresponding metadata (which refers to a customComponent) to the columnMetadata array.

In version 0.3.1, this works and I see the information from the customComponent come through in that column. But in version 0.5.1, I can see the heading for the extra column, but the column below is empty and the customComponent isn't rendered (inserting a console.log into the custom component proves the code is never hit as nothing is written to the console).

I don't remember this being documented anywhere at the time (it was very much a "hit and hope" scenario) but I thought I'd ask the question here in case it's known of, but I can't see any current issues open for it though, or if it's changed and I'm now doing something wrong?

**Code snippets:**

`<Griddle columns={["sku","category","description","price","basketOptions"]}`
`columnMetadata={[ ... removed other metadata for simplicity ...`
`{columnName:"basketOptions",displayName:"Options",customComponent:BasketOptions}]} />`

and the customComponent (reduced to just show static text for simplicity) is:

`class DummyColumnComponent extends React.Component {`
`render() {`
`return(`
`<div>text</div>`
`);`
`}`
`}`

**Comment by glifchits** [comment]
I'm also running into this. After upgrading to 0.5.1, a column is not rendering. Where there were 4 `<th>`s and 4 `<td>`s in each row before, now there are just 3 `<td>`s.

**Comment by ryanlanciaux** [comment]
Thank you for reporting this issue. I thought I understood what was going on but I think it may require slightly more digging to figure this out. Would it be possible for someone to make a rudimentary version of this on jsfiddle/codepen/jsbin etc?

@glifchits I want to make sure I understand this correctly -- in your table, you're using custom components with backing data for each column or do three of them have a corresponding record in the data and one is a custom component that doesn't correspond to an item in the data?

**Comment by glifchits** [comment]
@ryanlanciaux that's correct, the `results` prop looks like 
```javascript
{
  _id: "52571edffb4e13e0f",
  active: true,
  name: "Name",
  size: 143,
}
```
and my columns are `[ 'buttons', 'name', 'size', 'active' ]`. Buttons is not rendering. Each of these columns is rendered with a custom component.

**Comment by jonifen** [comment]
@ryanlanciaux I've created a jsbin here: [http://jsbin.com/buyaduqeji/1/edit?html,js,output](http://jsbin.com/buyaduqeji/1/edit?html,js,output)

**Comment by ryanlanciaux** [comment]
Thank you -- I'll try to get this resolved and released this weekend! Sorry about any problems that is causing.

**Comment by annyhe** [comment]
I am also getting the same issue: data in the custom column is not rendered. A fix would be appreciated.

**Comment by catdevnull** [comment]
I'm also experiencing this issue with 0.5.1! @ryanlanciaux any update on when this might be resolved? It's currently preventing me from updating to React 15 as I'm having to stay with Griddle 0.3.1 which requires React 14!

Cheers!

**Comment by modomodo** [comment]
+1

**Comment by SachaG** [comment]
Just ran into the same issue, downgrading to 0.3.1 for now.

**Comment by modomodo** [comment]
@SachaG Let  me know if the downgrade works. I tried downgrading too but to no avail

**Comment by SachaG** [comment]
Yes, downgrading worked for me.

**Comment by andrewmclagan** [comment]
Also experiencing this problem can we have this reopened

**Comment by M3lkior** [comment]
Same problem/needs here

**Comment by h0x91b** [comment]
I fount a workaround, need manually add on first element of `results` a missing field...

In my case columnName "actions" not exists in `results`, so

`this.state.users[0].actions = 'something';`

`<Griddle
	results={this.state.users}
	columnMetadata={columnMeta}
	showFilter={true}
	showSettings={true}
	useFixedHeader={true}
	columns={columns}
	onRowClick={this.onRowClick.bind(this)}
/>`

**Comment by jonifen** [comment]
Thanks for the workaround h0x91b. I did think of this initially, but wanted to avoid changing the data to avoid potential problems further down the line.
However, it looks like it might have been fixed as part of PR #454 - I expect it'll be in the next version made available on npm.

**Comment by jonifen** [comment]
Just realised I didn't close this once the later version was released (sorry!). Can't recreate the issue in 0.6.1, so I'll close this issue now.
